### PR TITLE
Fix for #38 and #36

### PR DESCRIPTION
**Pull request by kirill-konshin**
Support string/buffer bodies in Response constructor
Request should have Body methods like text(), json()

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -26,156 +26,13 @@ function Response(body, opts) {
...
- * Decode response as json
- *
- * @return  Promise
- */
-Response.prototype.json = function() {
 
-	return this._decode().then(function(text) {
-		return JSON.parse(text);
-	});
+	Body.call(this, body, opts);
```
Comment (TimothyGu) [inline lib/response.js]: You need at least a `Body.Promise = Response.Promise`

Comment (kirill-konshin) [inline lib/response.js]:
```js
// expose Promise
Response.Promise = global.Promise;
// expose Promise
Body.Promise = global.Promise;
```

It is exposed this way.

Comment (TimothyGu) [inline lib/response.js]: That's only for the default. The whole system is designed in which the module passes down its own `Promise` to the modules it calls.

**Comment by alubbe** [comment]
Any news on this? It'd be really helpful to have this functionality.

**Comment by kirill-konshin** [comment]
I have rebased the PR on top of most recent changes. But CI fails due to:

```
/home/travis/build/bitinn/node-fetch/node_modules/coveralls/bin/coveralls.js:18
        throw err;
              ^
Bad response: 500 {"message":"Build processing error.","error":true,"url":""}
```

I have nothing to do with it...

**Comment by bitinn** [comment]
I have no issue with it, I will fix the coverage when available.

**Comment by bitinn** [comment]
It's been a long road for this PR, thanks for sticking with it :)

**Comment by kirill-konshin** [comment]
No problem )

**Pull request merged**